base/server/src/main/java/com/netscape/cmscore/request/RequestRepository.java

```diff
@@ -357,61 +359,64 @@ public void removeAllObjects() throws EBaseException {
         }
     }
 
-    public RequestList listRequestsByFilter(String filter) throws EBaseException {
+    public Collection<RequestRecord> listRequestsByFilter(String filter) throws EBaseException {
 
+        Collection<RequestRecord> records = new ArrayList<>();
         DBSSession s = dbSubsystem.createSession();
-        DBSearchResults results = null;
 
         try {
-            results = s.search(mBaseDN, filter);
+            DBSearchResults sr = s.search(mBaseDN, filter);
+
+            while (sr.hasMoreElements()) {
+                RequestRecord record = (RequestRecord) sr.nextElement();
+                records.add(record);
+            }
 
```

Comment:
Since DBSearchResults is an Enumeration it could be converted to List using the static method `List<T> Collections.list(Enumeration<T>).

The same for the other cases below.

Reply:
Hmm.. I couldn't get it to compile since the `DBSearchResults` is an `Enumeration<Object>` and the method is returning `Collection<RequestRecord>`, so for now I think I'll keep the code as is, but maybe we can figure out something later.